src/commands/test/lib/espresso-preparer.ts

```diff
@@ -0,0 +1,149 @@
+import { IFileDescriptionJson } from "./test-manifest-reader";
+import { out } from "../../../util/interaction";
+import * as _ from "lodash";
+import * as fs from "fs";
+import * as path from "path";
+import * as glob from "glob";
+import * as pfs from "../../../util/misc/promisfied-fs";
+
+export class EspressoPreparer {
+  private readonly artifactsDir: string;
+  private buildDir: string;
+  private projectDir: string;
+  public include: IFileDescriptionJson[];
+  public testParameters: { [key:string]: any };
+
+  constructor(artifactsDir: string, projectDir?: string, buildDir?: string) {
+    if (!artifactsDir) {
+      throw new Error("Argument artifactsDir is required");
+    }
+
+    this.projectDir = projectDir;
+    this.buildDir = buildDir;
+    this.artifactsDir = artifactsDir;
+
+    this.validateEitherProjectOrBuildDir();
+  }
+
+  private validateEitherProjectOrBuildDir() {
+    if ((this.projectDir && this.buildDir) || !(this.projectDir || this.buildDir)) {
+      throw new Error("Either projectDir or buildDir must be specified");
+    }
+  }
+
+  public async prepare(): Promise<string> {
+    if (this.projectDir) {
+      await this.validateProjectDir();
+      this.buildDir = await this.generateBuildDirFromProject();
+    }
+
+    this.validateBuildDir();
+
+    await pfs.copyDir(this.buildDir, this.artifactsDir);
+
+    let manifestPath = path.join(this.artifactsDir, "test-manifest.json");
+    let manifest = await this.createEspressoManifest();
+    let manifestJson = JSON.stringify(manifest, null, 1);
+    await pfs.writeFile(manifestPath, manifestJson);
+
+    return manifestPath;
+  }
+
+  private async validateProjectDir() {
+    await this.validatePathExists(
+      this.projectDir, 
+      false, 
+      `Project directory ${this.projectDir} doesn't exist`);
+  }
+
+  private async generateBuildDirFromProject(): Promise<string> {
+    throw new Error("Not implemented");
+  }
+
+  private async validateBuildDir() {
+    await this.validateBuildDirExists();    
+    await this.validateTestApkExists();
+  }
+
+  private async globAsync(pattern: string): Promise<string[]> {
+    return new Promise<string[]>((resolve, reject) => {
+      glob(pattern, (err, matches) => {
+        if (err) {
+          reject(err);
+        }
+        else {
+          resolve(matches);
+        }
+      });
+    });
+  }
+
+  private async validateBuildDirExists() {
+    await this.validatePathExists(
+      this.buildDir,
+      false,
+      `Espresso build directory "${this.buildDir}" doesn't exist`);
+  }
+
+  private async validateTestApkExists(): Promise<void> {
+    await this.testApkPath();      
+  }
+
+  private async testApkPath(): Promise<string> {
```

Comment:
The "test prepare uitest" requires "--app-path" option that points to the application file. Should we do the same here?